src/hooks/index.ts

```diff
@@ -245,6 +248,39 @@ export const useTokenInfo = (
   return query.data
 }
 
+export const useNftModerationStatus = ({wallet, fingerprint}: {wallet: YoroiWallet; fingerprint: string | null}) => {
+  return useQuery({
+    queryKey: [wallet.id, 'nft', fingerprint],
+    queryFn: () => (fingerprint ? wallet.fetchNftModerationStatus(fingerprint) : null),
+    enabled: !!fingerprint,
+  })
+}
+
+export const useTokenImage = ({wallet, tokenId}: {wallet: YoroiWallet; tokenId: string}): string | null => {
+  const nft = useNft(wallet, {id: tokenId})
+  const fingerprint = nft ? getAssetFingerprint(nft.metadata.policyId, nft.metadata.assetNameHex) : null
+  const {data} = useNftModerationStatus({wallet, fingerprint})
+
+  if (nft && data === 'approved') {
+    return nft.image
+  }
+
+  return null
+}
+
+export const useNftImageModerated = ({
+  wallet,
+  nftId,
+}: {
+  wallet: YoroiWallet
+  nftId: string
+}): {image: string; status: YoroiNFTModerationStatus} | null => {
+  const nft = useNft(wallet, {id: nftId})
+  const fingerprint = nft ? getAssetFingerprint(nft.metadata.policyId, nft.metadata.assetNameHex) : null
```

Comment:
whats the difference between `nft.id` and `getAssetFingerprint(nft.metadata.policyId, nft.metadata.assetNameHex)`?